sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobUrlParts.java

```diff
@@ -296,7 +293,7 @@ public static BlobUrlParts parse(String url) {
     public static BlobUrlParts parse(URL url) {
         BlobUrlParts parts = new BlobUrlParts().setScheme(url.getProtocol());
 
-        if (IP_V4_URL_PATTERN.matcher(url.getHost()).find()) {
+        if (ModelHelper.IP_V4_URL_PATTERN.matcher(url.getHost()).find()) {
```

Comment:
I think the toUrl method may also need to be updated. It looks like it always sets the container right after the host and never considers if it needs to set the account name?

Reply:
Updated